src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternMatcher.cs

```diff
@@ -499,7 +496,7 @@ private bool SavePathSegmentsAsValues(int index, RouteValueDictionary values, St
             {
                 foreach (var item in outValues)
                 {
-                    values.Add(item.Key, item.Value);
+                    values[item.Key] = item.Value;
```

Comment:
This is a quirk of how 'slots' for route values work. You sometimes need to overwrite a value. This won't have any functional impact on the old code path.